core/bin/snapshots_creator/src/main.rs

```diff
@@ -0,0 +1,273 @@
+use anyhow::Context as _;
+use prometheus_exporter::PrometheusExporterConfig;
+use std::time::Duration;
+use tokio::sync::watch;
+use tokio::sync::watch::Receiver;
+use vise::{Buckets, Gauge, Histogram, Metrics};
+use zksync_config::configs::PrometheusConfig;
+use zksync_config::{PostgresConfig, SnapshotsCreatorConfig};
+
+use zksync_dal::ConnectionPool;
+use zksync_env_config::object_store::SnapshotsObjectStoreConfig;
+use zksync_env_config::FromEnv;
+use zksync_object_store::{ObjectStore, ObjectStoreFactory};
+use zksync_types::snapshots::{
+    SnapshotFactoryDependencies, SnapshotStorageLogsChunk, SnapshotStorageLogsStorageKey,
+};
+use zksync_types::zkevm_test_harness::zk_evm::zkevm_opcode_defs::decoding::AllowedPcOrImm;
+use zksync_types::{L1BatchNumber, MiniblockNumber};
+use zksync_utils::ceil_div;
+use zksync_utils::time::seconds_since_epoch;
+
+#[derive(Debug, Metrics)]
+#[metrics(prefix = "snapshots_creator")]
+struct SnapshotsCreatorMetrics {
+    storage_logs_chunks_count: Gauge<u64>,
+
+    storage_logs_chunks_left_to_process: Gauge<u64>,
+
+    snapshot_generation_duration: Gauge<u64>,
```

Comment:
It's very interesting to see completely different mapping of metrics to prometheus 'primitives' than what I'm used to. Cronjobs are very rarely run more often than every hour so for me using a histogram feels 'hacky' here. In such scenarios there is only one job run at the same time and we want only a single number to track. That being said, as those two primitives have comparable features, I totally buy the consistency argument (also because we don't and probably won't have many cronjobs)